lib/Doctrine/DBAL/TransactionManager.php

```diff
@@ -0,0 +1,284 @@
+<?php
+
+namespace Doctrine\DBAL;
+
+/**
+ * Manages transactions.
+ */
+class TransactionManager
+{
+    const SAVEPOINT_PREFIX = 'DOCTRINE2_SAVEPOINT_';
+
+    /**
+     * @var \Doctrine\DBAL\Connection
+     */
+    private $connection;
+
+    /**
+     * @var \Doctrine\DBAL\Transaction[]
+     */
+    private $activeTransactions = array();
+
+    /**
+     * Whether nested transactions should use savepoints.
+     *
+     * @var boolean
+     */
+    private $nestTransactionsWithSavepoints = false;
+
+    /**
+     * @param \Doctrine\DBAL\Connection $connection
+     */
+    public function __construct(Connection $connection)
+    {
+        $this->connection = $connection;
+    }
+
+    /**
+     * @return TransactionBuilder
+     */
+    public function createTransaction()
+    {
+        return new TransactionBuilder($this);
+    }
+
+    /**
+     * @param array $configuration
+     *
+     * @return \Doctrine\DBAL\Transaction
+     */
+    public function beginTransaction(array $configuration = array())
+    {
+        $this->connection->connect();
+
+        if (isset($configuration[TransactionBuilder::ISOLATION_LEVEL])) {
```

Comment:
With this approach, the manager depends on implementation details of the builder and transaction (particularly, the constant and the configuration as an array). Configuration as an array in the API looks like a bad idea to me.

Reply:
This also stems from the generic configuration idea. We'd definitely be better off with something like `$builder->getIsolationLevel()`.